### PR TITLE
feat(components): expose new components

### PR DESCRIPTION
**Pull request by marstamm**
This PR adds the newly exported Components and HOCs from https://github.com/camunda/camunda-modeler/pull/4568 to this helper.

related to https://github.com/camunda/camunda-modeler/pull/4568

### CODE REVIEW

File: components.js [hunk 1]
```diff
@@ -157,4 +157,100 @@ export const Section = window.components.Section || NotCompatible('5.0');
...
+ *            switcherLabel="Switcher label"
+ *            id={ id }
+ *            name={ name }
+ *            description="Toggle description"
+ *          />
+ *        </Form>
+ *       )}
+ *    </Formik>
+ *   );
+ * }
```
Comment (barmac) [inline components.js]: This needs to be updated.

Comment (barmac) [reply components.js]:
```suggestion
 * @example
 * 
 * import { TextInput } from 'camunda-modeler-plugin-helpers/components';
 *
 * function CustomInput(props) {
 *   return (
 *    <Formik initialValues={ initialValues } onSubmit={ this.onSubmit }>
 *      {() => (
 *        <Form>
 *          <Field
 *            component={ TextInput }
 *            label="My input"
 *            id={ id }
 *.           multiline={false}
 *            name={ name }
 *            description="Custom description"
 *          />
 *        </Form>
 *       )}
 *    </Formik>
 *   );
 * }
 ```

File: components.js [hunk 2]
```diff
@@ -157,4 +157,100 @@ export const Section = window.components.Section || NotCompatible('5.0');
...
  * }
  */
-export const ToggleSwitch = window.components.ToggleSwitch || NotCompatible('5.0');
\ No newline at end of file
+export const ToggleSwitch = window.components.ToggleSwitch || NotCompatible('5.0');
+
+ /**
+ * TextInput component.
+ *
+ * @type {import('react').ComponentType<{ hint: string, label: string, fieldError: string, multiline: boolean, description: string }>}
```
Comment (barmac) [inline components.js]: Isn't the `name` missing?

File: components.js [hunk 3]
```diff
@@ -157,4 +157,100 @@ export const Section = window.components.Section || NotCompatible('5.0');
...
+/**
+ * A helper function to create Tab components
+ * to be used with the TabProvider.
+ *
+ * @type {Function}
+ * @param {string} tabName - The name of the tab.
+ * @param {object} providers - The providers object.
+ *   @property {string} providers.type - The type of the provider.
+ *   @property {React.Component} providers.editor - The editor component.
+ *   @property {string} providers.defaultName - The default name of the provider.
```
Comment (barmac) [inline components.js]: This looks like an error. Shouldn't the parameter keys be specified as `@param` too? On the other hand, neither of the options works in VSCode...

**Comment by nikku** [comment]
@marstamm Is this still relevant?

**Comment by marstamm** [comment]
Yes, just did not have time to address the Review yet

File: components.js (hunk 3 above)
Comment (marstamm) [reply components.js]: `@param` works for me, I adjusted it in [df906e7](https://github.com/camunda/camunda-modeler-plugin-helpers/pull/26/commits/df906e74e56971c8b76adf359cb945316371cb21)

File: components.js (hunk 2 above)
Comment (marstamm) [reply components.js]: Yes

File: components.js (hunk 1 above)
Comment (marstamm) [reply components.js]: Added in [df906e7](https://github.com/camunda/camunda-modeler-plugin-helpers/pull/26/commits/df906e74e56971c8b76adf359cb945316371cb21)

File: components.js [hunk 4]
```diff
@@ -157,4 +157,101 @@ export const Section = window.components.Section || NotCompatible('5.0');
...
+ * function CustomInput(props) {
+ *   return (
+ *    <Formik initialValues={ initialValues } onSubmit={ this.onSubmit }>
+ *      {() => (
+ *        <Form>
+ *          <Field
+ *            component={ TextInput }
+ *            label="My input"
+ *            id={ id }
+ *.           multiline={false}
```
Comment (jarekdanielak) [inline components.js]:
```suggestion
 *            multiline={ false }
```

File: components.js [hunk 5]
```diff
@@ -157,4 +157,101 @@ export const Section = window.components.Section || NotCompatible('5.0');
...
+ /**
+ * CachedComponent class.
+ *
+ * @type {import('react').ComponentClass}
+ *
+ * @example
+ * 
+ * import { CachedComponent } from 'camunda-modeler-plugin-helpers/components';
+ * 
+ * class ComponentWithCachedStated extends CachedComponent {
```
Comment (jarekdanielak) [inline components.js]:
```suggestion
 * class ComponentWithCachedState extends CachedComponent {
```

**Review by jarekdanielak — APPROVED** [review]
Found small typos, but all good 👍

**Comment by marstamm** [comment]
Thank you for the review! Added and squashed 🎾

**Pull request merged**